src/confirm_subscription.rs

```diff
@@ -0,0 +1,74 @@
+#[ft_sdk::processor]
+pub fn confirm_subscription(
+    mut conn: ft_sdk::Connection,
+    ft_sdk::Query(code): ft_sdk::Query<"code">,
+    ft_sdk::Query(email): ft_sdk::Query<"email">,
+    ft_sdk::Query(next): ft_sdk::Query<"next", Option<String>>,
+) -> ft_sdk::processor::Result {
+    use diesel::prelude::*;
+
+    if !validator::ValidateEmail::validate_email(&email) {
+        return Err(ft_sdk::single_error("email", "Invalid email format.").into());
+    }
+
+    #[derive(diesel::QueryableByName)]
+    #[diesel(table_name = ft_sdk::auth::fastn_user)]
+    struct UserData {
+        id: i64,
+        data: String,
+    }
+
+    let (user_id, data): (i64, serde_json::Value)  = match diesel::sql_query(
+        r#"
+        SELECT id, data from fastn_user where json_extract(data, '$.subscription.confirmation_key') = $1 LIMIT 1;
+        "#,
+    )
```

Comment:
I think we should use same structure as `ft_sdk::auth::ProviderData` so that we can use `ft_sdk::auth::provider::user_data_by_custom_attribute` etc to do read and write operations.

Reply:
how we store auth data is going to change definitely. We've agreed that the current way of storing everything as json is not working great for us. This will definitely change then but now